core/attributedef.class.inc.php

```diff
@@ -5888,14 +5888,31 @@ public function GetForJSON($value)
 	public function GetAllowedValues($aArgs = array(), $sContains = '')
 	{
 		$aRawValues = parent::GetAllowedValues($aArgs, $sContains);
-		if (is_null($aRawValues))
-		{
+		if (is_null($aRawValues)) {
 			return null;
 		}
-		$aLocalizedValues = array();
-		foreach($aRawValues as $sKey => $sValue)
-		{
-			$aLocalizedValues[$sKey] = $this->GetValueLabel($sKey);
+		$aLocalizedValues = [];
+		$aSortedValues = [];
+		//bHaveSortedValues : some values have rank in xml
+		$bHaveSortedValues = false;
+		foreach ($aRawValues as $sKey => $sValue) {
+			$aVal = explode('::', $sValue);
+			if (count($aVal) == 1) {
+				$aLocalizedValues[$sValue] = $this->GetValueLabel($sValue);
+			} else {
+				$aSortedValues[$aVal[1]] = $aVal[0];
+				$bHaveSortedValues = true;
+			}
+		}
+		asort($aLocalizedValues);
+		
+		if ($bHaveSortedValues) {
```

Comment:
I don't think this boolean is needed:
```suggestion
		if (!empty($aSortedValues)) {
```